src/date/month-names.js

```diff
@@ -0,0 +1,82 @@
+define( function() {
+
+/**
+ * monthNames( months, type, length, calendar, cldr )
+ *
+ * @months [Object]
+ * Associative array of month indices to names. E.g. {"1": "January", "2": "February", etc.}
+ * This may be null for numeric months (length <=2). In this case, if necessary, the array
+ * will be synthesized.
+ *
+ * @type [String]
+ * Either "M" for formatted month names or "L" for stand-alone month names
+ *
+ * @length [Number]
+ * Size of the month field
+ * ref: http://www.unicode.org/reports/tr35/tr35-dates.html#Date_Format_Patterns
+ *
+ * @calendar [String]
+ * Name of the calendar system (e.g. "Gregorian", "Islamic")
+ *
+ * @cldr [Cldr instance].
+ *
+ * Returns a modified months array to include leap months in a format that allows for easier parsing
+ * and formatting, and works with GDate.
+ *
+ */
+return function( months, type, length, calendar, cldr ) {
+	var i, month, monthParts, monthPatterns, pattern;
+
+	if ( months ) {
+
+	// fastest (native) way to clone an object
+		months = JSON.parse( JSON.stringify( months ) );
+	}
```

Comment:
are you talking about the clone?  `months` is returned from `cldr.main('dates/calendar')` and is the actual data from the global CLDR object. I'm hesitant to change that directly; I don't know if any other part of the program relies on the month names in their original form.
You know cldr.js; if it's OK to modify it, we can drop that line.